ddtrace/internal/coverage/instrumentation_py3_10.py

```diff
@@ -0,0 +1,314 @@
+            if original_offset in line_starts:
+                # Inject trap call at the beginning of the line. Keep track
+                # of location and size of the trap call instructions. We
+                # need this to adjust the location table.
+                line = line_starts[original_offset]
+                trap_instructions = trap_call(trap_index, len(new_consts))
+                traps[original_offset] = len(trap_instructions)
+                instructions.extend(trap_instructions)
+                new_consts.append((line, trap_arg))
+
+                line_map[original_offset] = trap_instructions[0]
+
+                seen_lines.add(line)
+
+            _, arg = next(code_iter)
+
+            offset = len(instructions) << 1
+
+            # Propagate code
+            instructions.append(Instruction(original_offset, opcode, arg))
+
+            # Collect branching instructions for processing
+            if opcode in AJump.__opcodes__:
+                jumps[offset] = AJump(original_offset, [*ext, arg])
+            elif opcode in RJump.__opcodes__:
+                jumps[offset] = RJump(original_offset, [*ext, arg], JumpDirection.from_opcode(opcode))
+
+            if opcode is EXTENDED_ARG:
+                ext.append(arg)
+            else:
+                ext.clear()
+    except StopIteration:
+        pass
+
+    # Collect all the old jump start and end offsets
+    jump_targets = {_ for j in jumps.values() for _ in (j.start, j.end)}
+
+    # Adjust all the offsets and map the old offsets to the new ones for the
+    # jumps
+    for index, instr in enumerate(instructions):
+        new_offset = index << 1
+        if instr.offset in jump_targets:
+            offset_map[instr.offset] = new_offset
+        instr.offset = new_offset
+
+    # Adjust all the jumps, neglecting any EXTENDED_ARGs for now
+    branches: t.List[Branch] = []
+    for jump in jumps.values():
+        new_start = offset_map[jump.start]
+        new_end = offset_map[jump.end]
+
+        # If we are jumping at the beginning of a line, jump to the
+        # beginning of the trap call instead
+        target_instr = line_map.get(jump.end, instructions[new_end >> 1])
+        branch: Branch = (
+            RBranch(instructions[new_start >> 1], target_instr)
+            if isinstance(jump, RJump)
+            else ABranch(instructions[new_start >> 1], target_instr)
+        )
+        target_instr.targets.append(branch)
+
+        branches.append(branch)
+
+    # Process all the branching instructions to adjust the arguments. We
+    # need to add EXTENDED_ARGs if the argument is too large.
+    process_branches = True
+    exts: t.List[t.Tuple[Instruction, int]] = []
+    while process_branches:
+        process_branches = False
+        for branch in branches:
+            jump_instr = branch.start
+            new_arg = branch.arg
+            jump_instr.arg = new_arg & 0xFF
+            new_arg >>= 8
```

Comment:
Dumb question, where did the 8 and 0xFF come from? Could these be consts maybe?

Reply:
No the explanation makes sense. I just haven't done this stuff in a while so... This was a clarification I just personally needed 😅